src/qt/subscribedialog.cpp

```diff
@@ -81,81 +81,125 @@ void SubscribeDialog::on_buttonBox_accepted()
     //SSub->m_attributes.qos.m_timeBasedFilter.hasChanged = true;
     SSub->m_attributes.qos.m_presentation.hasChanged = true;
 
-    //Reliability
-    if(this->ui->checkBox_reliable->isChecked())
+    //RELIABILITY
+    if (this->ui->checkBox_reliable->isChecked())
+    {
         SSub->m_attributes.qos.m_reliability.kind = RELIABLE_RELIABILITY_QOS;
+    }
 
     //DURABILITY
     // cout << "Durability INDEX: "<< this->ui->comboBox_durability->currentIndex() << endl;
-    switch(this->ui->comboBox_durability->currentIndex())
+    switch (this->ui->comboBox_durability->currentIndex())
+    {
+        case 0: SSub->m_attributes.qos.m_durability.kind = VOLATILE_DURABILITY_QOS; break;
+        case 1: SSub->m_attributes.qos.m_durability.kind = TRANSIENT_LOCAL_DURABILITY_QOS; break;
+    }
+
+    //LIVELINESS
+    // cout << "LIVELINESS "<<this->ui->comboBox_liveliness->currentIndex()<<endl;
+    if (this->ui->comboBox_liveliness->currentIndex() == 0)
     {
-    case 0: SSub->m_attributes.qos.m_durability.kind = VOLATILE_DURABILITY_QOS; break;
-    case 1: SSub->m_attributes.qos.m_durability.kind = TRANSIENT_LOCAL_DURABILITY_QOS; break;
+       SSub->m_attributes.qos.m_liveliness.kind = AUTOMATIC_LIVELINESS_QOS;
     }
-    //Ownership:
-    switch(this->ui->comboBox_ownership->currentIndex())
+    if (this->ui->comboBox_liveliness->currentIndex() == 1)
     {
-    case 0: SSub->m_attributes.qos.m_ownership.kind = SHARED_OWNERSHIP_QOS; break;
-    case 1:
+       SSub->m_attributes.qos.m_liveliness.kind = MANUAL_BY_PARTICIPANT_LIVELINESS_QOS;
+    }
+    if (this->ui->comboBox_liveliness->currentIndex() == 2)
     {
-        SSub->m_attributes.qos.m_ownership.kind = EXCLUSIVE_OWNERSHIP_QOS;
-        SSub->m_shapeHistory.m_isExclusiveOwnership = true;
-        break;
+       SSub->m_attributes.qos.m_liveliness.kind = MANUAL_BY_TOPIC_LIVELINESS_QOS;
     }
+    if (this->ui->lineEdit_leaseDuration->text()=="INF")
+    {
+       SSub->m_attributes.qos.m_liveliness.lease_duration = c_TimeInfinite;
     }
-    // Deadline
-    if(this->ui->lineEdit_Dead->text()=="INF")
+    else
+    {
+        QString value = this->ui->lineEdit_leaseDuration->text();
+        if (value.toDouble()>0)
+        {
+            SSub->m_attributes.qos.m_liveliness.lease_duration = rtps::TimeConv::MilliSeconds2Time_t(value.toDouble()).to_duration_t();
+            SSub->m_attributes.qos.m_liveliness.announcement_period =
+                rtps::TimeConv::MilliSeconds2Time_t(value.toDouble()/2).to_duration_t();
```

Comment:
It would be better if we added a text edit for the announcement period. This way the user can configure everything that is related to liveliness

Reply:
Now I see that I made this comment in the subscriber GUI. Actually the announcement period only applies to the publisher, so it should be displayed in the publisher dialog but not in the subscriber one. Can we remove it from here and keep it for the publisher, please?